### ISSUE TITLE
Programmierkonventionen

### ISSUE THREAD

**Issue by funketh** [rrickfox/traffic-sim#3]
[**Unity Coding Guidelines**](https://avangarde-software.com/unity-coding-guidelines-basic-best-practices/)

Dem hinzufügen würde ich vielleicht:
- Private Attribute: `_trafficSimulator` (Unterstrich vor dem Variablennamen)

_Edit: Update link_

**Comment by RichardKarlFuchs** [comment]
Ich würde auch noch:
- Konstanten: `TIME_STEP` (Großbuchstaben und Unterstrich)

Eigentlich ist es `TimeStep` aber das ist leichter erkennbar und führt zu keinen Verwechslungen mit Methoden

**Comment by RichardKarlFuchs** [comment]
- Wenn begrenzt viele Werte: `enum` statt `String`

**Comment by funketh** [comment]
Nicht wirklich programmieren, aber ähnliche Commits sollten gesquashed werden, um Übersichtlichkeit beizubehalten.
Branches, die nur sehr ähnliche Commits enthalten, sollten durch squashen gemerged werden.

**Comment by rrickfox** [comment]
Soll aber nicht heißen, dass alle Branches durch squashen gemerged werden, richtig?

**Comment by funketh** [comment]
Doch, denke schon. Der Punkt ist, dass es immer viel zu viele Commits gibt, um ihnen sinnvolle Informationen zu entnehmen. Außerdem arbeiten ältere Commits häufig mit nem ganz anderen Datenmodell als neuere, was sie ziemlich sinnlos macht.

Der gesquashte Commit kann ja sehr ausführlich sein. Dann versteht man auch wenigstens alle Änderungen eines Pull Requests durch lesen einer Commit-Nachricht.

**Comment by funketh** [comment]
Es wurde sich geeinigt, dass bei Pull Requests immer normale Merges ausgeführt werden.
Wichtig ist jedoch, dass die dem Merge beigefügte Nachricht **sämtliche Commits des Branches zusammenfasst**, sodass man sich nicht die einzelnen Commits durchlesen muss.
